numba_cuda/numba/cuda/testing.py

```diff
@@ -203,3 +207,75 @@ class ForeignArray(object):
     def __init__(self, arr):
         self._arr = arr
         self.__cuda_array_interface__ = arr.__cuda_array_interface__
+
+
+def filecheck_ir(
+    ir_content: str,
+    check_patterns: str,
+    check_prefixes: list[str] = ["CHECK"],
+    **extra_filecheck_options: dict[str, str | int],
+) -> bool:
```

Comment:
If it always succeeds or raises an `AssertionError`, then there's probably no point returning anything.

Reply:
Good catch, I forgot to update the return value and type annotations after making a change. Updated!